multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -309,33 +309,71 @@ func (c *Cluster) validateKeyPair(ec2Svc ec2Service) error {
 type r53Service interface {
 	ListHostedZonesByName(*route53.ListHostedZonesByNameInput) (*route53.ListHostedZonesByNameOutput, error)
 	ListResourceRecordSets(*route53.ListResourceRecordSetsInput) (*route53.ListResourceRecordSetsOutput, error)
+	GetHostedZone(*route53.GetHostedZoneInput) (*route53.GetHostedZoneOutput, error)
 }
 
 func (c *Cluster) validateDNSConfig(r53 r53Service) error {
 	if !c.CreateRecordSet {
 		return nil
 	}
 
-	zonesResp, err := r53.ListHostedZonesByName(&route53.ListHostedZonesByNameInput{
-		DNSName: aws.String(c.HostedZone),
-	})
+	if c.HostedZoneID == "" {
+		//TODO(colhom): When HostedZone parameter is gone, this block can be removed
+		//Config will gaurantee that HostedZoneID is set from the get-go
+		listHostedZoneInput := route53.ListHostedZonesByNameInput{
+			DNSName: aws.String(c.HostedZone),
+		}
+
+		zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
+		if err != nil {
+			return fmt.Errorf("Error validating HostedZone: %s", err)
+		}
+
+		zones := zonesResp.HostedZones
+
+		if len(zones) == 0 {
+			return fmt.Errorf("hosted zone %s does not exist", c.HostedZone)
+		}
+
```

Comment:
Unfortunately `ListHostedZonesByNameInput.DNSName` only controls lexical ordering of results. The way this api call works is very weird, but tldr for this use case is: 

you'll get a list that has two contiguous segments. It starts with all the hosted zones that match `DNSName` exactly followed by hosted zones which don't. Either or both segments may be zero length.

I get the feeling they just wrapped their fuzzy search api for this functionality or something crazy like that.